hazelcast-client-legacy/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxyBase.java

```diff
@@ -201,6 +201,10 @@ public void run() {
                 if (completionListener != null) {
                     completionListener.onException(e);
                 }
+            } catch (Throwable t) {
```

Comment:
@sancar 

Even though there is eviction, force eviction and cross partition (but owned by same partition thread) based force eviction, still there might be `NativeOutOfMemoryError`.

The simplest case is that for example all native memory occupied by *parition_thread_1*, .., *parition_thread_n-1* and *partition_thread_n* has not used any native memory yet because there was no operation targeted to partitions owned by *partition_thread_n*. And then an operation comes to partition owned by *partition_thread_n* but all native memory is already in use. Also since there is no used native memory by neither that partition nor that partition thread, all evictions will have no effect so in this case, there will be `NativeOutOfMemoryError`. 

In this case, `CompletionListener` should be notified since `NativeOutOfMemoryError` might occur because there is no %100 guarantee